// qnui/lib/_components/@alife/next-timeline/lib/view/timeline.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports['default'] = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _nextUtil = require('../../../next-util/lib/index.js');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _defaults(obj, defaults) { var keys = Object.getOwnPropertyNames(defaults); for (var i = 0; i < keys.length; i++) { var key = keys[i]; var value = Object.getOwnPropertyDescriptor(defaults, key); if (value && value.configurable && obj[key] === undefined) { Object.defineProperty(obj, key, value); } } return obj; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : _defaults(subClass, superClass); }

var Timeline = (_temp = _class = function (_Component) {
    _inherits(Timeline, _Component);

    function Timeline(props) {
        _classCallCheck(this, Timeline);

        var _this = _possibleConstructorReturn(this, _Component.call(this, props));

        _this.state = {
            fold: props.fold
        };
        return _this;
    }

    Timeline.prototype.componentWillReceiveProps = function componentWillReceiveProps(newProps) {
        this.setState({
            fold: newProps.fold
        });
    };

    Timeline.prototype.toggleFold = function toggleFold(folderIndex, total) {
        var fold = this.state.fold;


        if (folderIndex) {
            for (var i = 0; i < fold.length; i++) {
                var _fold$i = fold[i],
                    foldArea = _fold$i.foldArea,
                    foldShow = _fold$i.foldShow;


                if (foldArea[1] && folderIndex === foldArea[1] || !foldArea[1] && folderIndex === total - 1) {
                    fold[i].foldShow = !foldShow;
                }
            }

            this.setState({
                fold: fold
            });
        }
    };

    Timeline.prototype.render = function render() {
        var _this2 = this,
            _classNames;

        var _props = _extends({}, this.props),
            prefix = _props.prefix,
            className = _props.className,
            others = _objectWithoutProperties(_props, ['prefix', 'className']);

        var children = others.children;

        var total = children.length;
        var pickProps = (0, _nextUtil.pickAttrs)(others);
        var fold = this.state.fold;

        var cloneChildren = [];
        var hasLeft = false;

        // 修改子节点属性
        _react.Children.map(children, function (child, i) {
            var folderIndex = null;
            var foldNodeShow = false;

            fold.forEach(function (item) {
                var foldArea = item.foldArea,
                    foldShow = item.foldShow;


                if (foldArea[0] && i >= foldArea[0] && (i <= foldArea[1] || !foldArea[1])) {
                    folderIndex = foldArea[1] || total - 1;
                    foldNodeShow = foldShow;
                }
            });

            var _child$props = _extends({}, child.props),
                left = _child$props.left,
                right = _child$props.right;

            hasLeft = hasLeft || left;

            var node = _react2['default'].cloneElement(child, {
                key: 'timeline-item-' + i,
                prefix: prefix,
                total: total,
                index: i,
                type: 'normal',
                folderIndex: folderIndex,
                foldShow: foldNodeShow
            });
            cloneChildren.push(node);

            if (folderIndex === i) {
                var _node = _react2['default'].cloneElement(child, {
                    key: 'timeline-item-fold-' + i,
                    prefix: prefix,
                    total: total,
                    index: i,
                    type: 'fold',
                    folderIndex: folderIndex,
                    foldShow: foldNodeShow,
                    toggleFold: _this2.toggleFold.bind(_this2, folderIndex, total)
                });
                cloneChildren.push(_node);
            }
        });

        var timelineCls = (0, _classnames2['default'])((_classNames = {}, _defineProperty(_classNames, prefix + 'timeline', true), _defineProperty(_classNames, prefix + 'timeline-no-left', !hasLeft), _defineProperty(_classNames, className, className), _classNames));

        return _react2['default'].createElement(
            'div',
            _extends({}, pickProps, { className: timelineCls }),
            _react2['default'].createElement(
                'ul',
                null,
                cloneChildren
            )
        );
    };

    return Timeline;
}(_react.Component), _class.propTypes = {
    prefix: _react.PropTypes.string,
    fold: _react.PropTypes.array, // [{foldArea: [3, 4], foldShow: true}]
    className: _react.PropTypes.string
}, _class.defaultProps = {
    prefix: 'next-',
    fold: []
}, _temp);
Timeline.displayName = 'Timeline';
exports['default'] = Timeline;
module.exports = exports['default'];
