// qnui/lib/_components/@alife/next-navigation/lib/util/helper.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var navigationPropTypes = {
    rootNavigation: _react.PropTypes.any,
    rootMount: _react.PropTypes.bool,
    navigation: _react.PropTypes.any,
    branchPadding: _react.PropTypes.any,
    prefix: _react.PropTypes.string,
    type: _react.PropTypes.string,
    leaf: _react.PropTypes.string,
    activeDirection: _react.PropTypes.string,
    contentAlign: _react.PropTypes.string,
    trigger: _react.PropTypes.string,
    title: _react.PropTypes.string,
    hasLeaf: _react.PropTypes.bool,
    selectedStyle: _react.PropTypes.bool,
    isStopPropagation: _react.PropTypes.bool,
    isBlurDispear: _react.PropTypes.bool,
    onItemClick: _react.PropTypes.func,
    onItemMouseEnter: _react.PropTypes.func,
    onItemMouseLeave: _react.PropTypes.func,
    onItemMouseMove: _react.PropTypes.func,
    onItemSelect: _react.PropTypes.func,
    onItemFold: _react.PropTypes.func,
    onItemUnFold: _react.PropTypes.func,
    getRootState: _react.PropTypes.func,
    selectedKey: _react.PropTypes.any,
    openedKeys: _react.PropTypes.array,
    accordion: _react.PropTypes.bool,
    branchLevel: _react.PropTypes.number
};

var helper = {
    propTypes: navigationPropTypes,
    empty: function empty() {}
};

exports['default'] = helper;
module.exports = exports['default'];
